### DISCUSSION TITLE
How to trigger a task group using TriggerDagRunOperator?

### DISCUSSION THREAD

**Discussion by Raul824** [apache/airflow#30985] [Q&A] [answered]
I was using subdags as we had 961 dags which were grouped to 52 dags using subdags. There were use cases when only 1 subdag needed to run from another dag which was easy by just putiing dag_nm.subdag_nm in the trigger_dag_id. Now we want to move to task groups but I don't know how to trigger only a single task group in a dag with multiple task groups from another dag.

elaborating it further.

DAG_KEY_1 has 13 task groups.
DAG_SCD_1 has 13 task groups.

if task_group 1 of DAG_KEY_1 is completed i need to trigger only task_group_1 of DAG_SCD_1 other 12 should not run.
This was previously being achieved by triggering subdag.

**Comment by potiuk** [comment, marked as answer]
You can't. If you want to have the possibility of independently scheduling your "smaller" dags you likely better convert your dags to dataset aware scheduling https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/datasets.html

This has multiple advantages, including the fact that each DAG and subdag of yours i a separate "micro-pipeline" you can also read more about the concept in https://www.astronomer.io/blog/micropipelines-a-microservice-approach-for-dag-authoring-in-apache-airflow/. Also this effectively annotates your DAGs with datasets, and that will have much better lineage properties once you use open-lineage for example.

**Reply by Raul824** [reply]
Would it be able to handle 1000+ dags with inter dependencies.
As I am migrating control M to airflow and have built a parser where if there are more than one dependency am using external task sensor and one to one dependency than trigger dag run operator.


Example dag from my parser using subdags. As the subdags has dependencies from within the dag

![dag_img](https://user-images.githubusercontent.com/51978647/235448997-e812e36a-cc98-4151-8909-7886595b44b0.png)

When you zoom into subdag the subdag has external dependencies as well.

![sub_dag_img](https://user-images.githubusercontent.com/51978647/235449000-40c4b684-9d22-4d3f-84da-567f3fe1a3f5.png)

So can I replace all external task sensors and trigger operator to use this dataset instead.

**Reply by potiuk** [reply]
I cannot answer 100% yes. I have not even implemented the dataset feature, but It's been implemented with performance in mind. 

Also - this is open-source software that you paid 0 USD for and as such it comes with zero guarantees that you can sue someone or take someone's word for granted and base your business decisions on.

But on the other hand - this is also an open-source software that almost 2500 people contributed to and you have a chance now to spend a little time to convert your (apparently big) use cases and by testing if it works, you could tell EVERYONE else that "yes, it can handle the scale". Or maybe try it out and see that it does not, report it and contribute back by finding some performance bottlenecks.

Are you up to the challenge and giving back for the countiess hours of volunteers who developed airflow so that you can use it for free and tell us if it is good enough to handle your case?

I am all ears.
